### PR TITLE
Major version update of the Datadog chart to 3.0

### PR DESCRIPTION
**Pull request by CharlyF**
#### What this PR does / why we need it:

- The Kubernetes State Metrics Core check is now enabled by default
- Kubernetes State Metrics (legacy) is off by default
- Deprecation notice for users updating to 3.0 and not migrating to Kubernetes State Metrics Core
- Minimum version of the Datadog Cluster Agent and the Datadog Agent updated
- Dropping support of Helm 2.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/README.md [hunk 1]
```diff
@@ -550,6 +496,7 @@ helm install --name <RELEASE_NAME> \
 | clusterAgent.envFrom | list | `[]` | Set environment variables specific to Cluster Agent from configMaps and/or secrets |
 | clusterAgent.healthPort | int | `5556` | Port number to use in the Cluster Agent for the healthz endpoint |
 | clusterAgent.image.digest | string | `""` | Cluster Agent image digest to use, takes precedence over tag if specified |
+| clusterAgent.image.doNotCheckTag | string | `nil` | Skip the version<>chart compatibility check |
```
Comment (maycmlee) [inline charts/datadog/README.md:499]:
```suggestion
| clusterAgent.image.doNotCheckTag | string | `nil` | Skip the version and chart compatibility check |
```

File: charts/datadog/values.yaml [hunk 2]
```diff
@@ -712,6 +712,15 @@ clusterAgent:
     pullSecrets: []
     #   - name: "<REG_SECRET>"
 
+    # clusterAgent.image.doNotCheckTag -- Skip the version<>chart compatibility check
+
+    ## By default, the version passed in clusterAgent.image.tag is checked
+    ## for compatibility with the version of the chart.
+    ## This boolean permits to completely skip this check.
```
Comment (maycmlee) [inline charts/datadog/values.yaml:719]:
```suggestion
    ## This boolean permits completely skipping this check.
```

File: charts/datadog/values.yaml [hunk 3]
```diff
@@ -712,6 +712,15 @@ clusterAgent:
...
+    ## This is useful, for example, for custom tags that are not
+    ## respecting semantic versioning
```
Comment (maycmlee) [inline charts/datadog/values.yaml:721]:
```suggestion
    ## respecting semantic versioning.
```

**Review by maycmlee — COMMENTED** [review]
Left some feedback for docs!

File: charts/datadog/README.md [hunk 4]
```diff
@@ -157,6 +123,12 @@ You can specify the Datadog Cluster Agent token used to secure the communication
 
 ### Upgrading
 
+#### From 2.x to 3.x
+
+The migration from 2.x to 3.x does not require manual action.
+As per the Changelog, we do not be guaranteeing support of Helm 2 moving forward.
+If you already have the Legacy Kubernetes State Metrics Check enabled, migrating will only show you the deprecation notifice.
```
[the remark this answers is not in the document]
Comment (CharlyF) [reply charts/datadog/README.md:130]: that's correct

**Comment by CharlyF** [comment]
@maycmlee - I could not commit your changes directly as I had to update the tmpl and run the doc generator, but I [incorporated the suggestions](https://github.com/DataDog/helm-charts/pull/738/commits/dcf95fbd5ed9c9d29881f9d175754614e7762ce2)
Thanks

File: charts/datadog/README.md [hunk 5]
```diff
@@ -29,7 +29,6 @@ Kubernetes 1.10+ or OpenShift 3.10+, note that:
 | Repository | Name | Version |
 |------------|------|---------|
 | https://helm.datadoghq.com | datadog-crds | 0.4.7 |
-| https://prometheus-community.github.io/helm-charts | kube-state-metrics | 2.13.2 |
```
Comment (clamoriniere) [inline charts/datadog/README.md:32]: is it attended that we remove this line even if we just disable the default installation?

File: charts/datadog/README.md.gotmpl [hunk 6]
```diff
@@ -57,22 +50,13 @@ After a few minutes, you should see hosts and metrics being reported in Datadog.
 
 **Note:** You can set your [Datadog site](https://docs.datadoghq.com/getting_started/site) using the `datadog.site` field.
 
-For Helm 3:
```
Comment (clamoriniere) [inline charts/datadog/README.md.gotmpl:60]: should we add a line to say that only helm 3 is supported.

File: charts/datadog/README.md.gotmpl [hunk 7]
```diff
@@ -152,6 +118,12 @@ You can specify the Datadog Cluster Agent token used to secure the communication
 
 ### Upgrading
 
+#### From 2.x to 3.x
+
+The migration from 2.x to 3.x does not require manual action.
+As per the Changelog, we do not be guaranteeing support of Helm 2 moving forward.
```
Comment (clamoriniere) [inline charts/datadog/README.md.gotmpl:124]: Ok the line that I mentioned earlier is here :)

File: charts/datadog/requirements.yaml [hunk 8]
```diff
@@ -5,7 +5,3 @@ dependencies:
     condition: clusterAgent.metricsProvider.useDatadogMetrics
     tags:
     - install-crds
-  - name: kube-state-metrics
-    version: 2.13.2
-    repository: https://prometheus-community.github.io/helm-charts
-    condition: datadog.kubeStateMetricsEnabled
```
Comment (clamoriniere) [inline charts/datadog/requirements.yaml:11]: the idea was to keep optional the installation on the `kube-state-metrics` chart dep.

**Review by clamoriniere — COMMENTED** [review]
few comments

File: charts/datadog/requirements.yaml [hunk 9]
```diff
@@ -8,4 +8,4 @@ dependencies:
   - name: kube-state-metrics
     version: 2.13.2
     repository: https://prometheus-community.github.io/helm-charts
-    condition: datadog.kubeStateMetricsEnabled
+    condition: datadog.kubeStateMetricsEnabled
```
Comment (clamoriniere) [inline charts/datadog/requirements.yaml:11]:
```suggestion
    condition: datadog.kubeStateMetricsEnabled

```

File: charts/datadog/values.yaml [hunk 10]
```diff
@@ -119,7 +119,7 @@ datadog:
   # datadog.kubeStateMetricsEnabled -- If true, deploys the kube-state-metrics deployment
 
   ## ref: https://github.com/kubernetes/kube-state-metrics/tree/kube-state-metrics-helm-chart-2.13.2/charts/kube-state-metrics
-  kubeStateMetricsEnabled: true
+  kubeStateMetricsEnabled: false
```
Comment (clamoriniere) [inline charts/datadog/values.yaml:122]: you can added a comment that in version 4.0.0 this option will be removed

**Review by clamoriniere — COMMENTED** [review]
2 small comments otherwise looks good

quick question: do we want to add the remove `datadog.apm.enabled` to?

**Review by clamoriniere — APPROVED** [review]
I'm looking forward to see this PR merged. 💯

**Comment by kennonkwok** [comment]
🎉

**Pull request merged**